File: Problems/1716. Calculate Money in Leetcode Bank/1716. Calculate Money in Leetcode Bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        w = n // 7; # Passed weeks.
        l = n % 7; # Remain days.
        res = 21*w + 7*sum(range(w+1));
        
        res += l*w + sum(range(l+1));
        
        return res;

File: Problems/1716. Calculate Money in Leetcode Bank/1716. Calculate Money in Leetcode Bank.cpp
class Solution {
public:
    int totalMoney(int n) {
        int w = n / 7; // Passed weeks.
        short l = n % 7; // Remain days.
        int res = 21*w + 7*(w*(w+1))/2;

        res += l*w + l*(l+1)/2;

        return res;
    }
};